test/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Tests/LibuvOutputConsumerTests.cs

```diff
@@ -382,16 +382,161 @@ public void Dispose()
                 Assert.False(task4Success.IsFaulted);
 
                 // Third task is now canceled
-                // TODO: Cancellation isn't supported right now
-                // await Assert.ThrowsAsync<TaskCanceledException>(() => task3Canceled);
-                // Assert.True(task3Canceled.IsCanceled);
```

Comment:
You should also be able to remove `var task4Success = socketOutput.WriteAsync(fullBuffer, default(CancellationToken));` since you don't advance any number of bytes on a write failure any more.